lightkurve/periodogram.py

```diff
@@ -767,10 +767,6 @@ def from_lightcurve(lc, minimum_frequency=None, maximum_frequency=None,
             raise ValueError('You have input keyword arguments for both frequency and period. '
                              'Please only use one.')
 
-        if (~np.isfinite(lc.flux)).any():
-            raise ValueError('Lightcurve contains NaN values. Use lc.remove_nans()'
-                             ' to remove NaN values from a LightCurve.')
-
```

Comment:
As the NaNs are already filtered out in the caller, there is no real need to remove the exception here. And worse, calling `LombScarglePeriodogram.from_lightcurve()` directly will now return an invalid periodogram without warning:
```
>>> pg = lk.periodogram.LombScarglePeriodogram.from_lightcurve(lc.normalize(unit='ppm'))
>>> pg.power
<Quantity [nan, nan, nan, ..., nan, nan, nan] ppm>
```